### ISSUE TITLE
[BUG] 0 in some cases for decimal being cast to a string returns different results.

### ISSUE THREAD

**Issue by revans2** [NVIDIA/spark-rapids#6339]
**Describe the bug**
This is related to https://github.com/NVIDIA/spark-rapids/issues/6337

after applying the patch we have two tests that fail.

```
FAILED ../../src/main/python/cast_test.py::test_cast_map_to_string[true-Map(Decimal(not_null)(7,7),Decimal(12,2))]
FAILED ../../src/main/python/cast_test.py::test_cast_map_to_string[false-Map(Decimal(not_null)(7,7),Decimal(12,2))]
```

It appears that they are failing because 0 is turned into `0.0000000` on the GPU but `0E-7` on the CPU.

**Comment by amahussein** [comment]
Spark-3.4 addressed this as bug

- https://github.com/NVIDIA/spark-rapids/issues/6015
- https://github.com/apache/spark/commit/c621df269d
- Followup: https://github.com/apache/spark/commit/5bd47640dc

**Comment by hyperbolic2346** [comment]
Can this be closed now that #8481 has been merged?
